src/main/java/dk/bec/polonez/reservationsystem/service/OfferService.java

```diff
@@ -35,36 +44,86 @@ public OfferService(OfferRepository offerRepository, FeatureRepository featureRe
     }
 
     public OfferDto getById(long id) {
-        Offer offer = offerRepository.getById(id);
-        return modelMapper.map(offer, OfferDto.class);
+        Optional<Offer> offer = offerRepository.findById(id);
+        return modelMapper.map(offer.orElseThrow(
+                () -> new ResponseStatusException(HttpStatus.NOT_FOUND, "Offer with id=" + id + " not found")), OfferDto.class);
     }
 
     public OfferDto addOffer(CreateOfferDto offerDto) {
-        Offer offer = modelMapper.map(offerDto, Offer.class);
-        Offer savedOffer = offerRepository.save(offer);
-        return modelMapper.map(savedOffer, OfferDto.class);
+        User currentUser = authService.getCurrentUser();
+        Role currentUserRole = currentUser.getRole();
+        if (currentUserRole.hasOfferCreatePrivilege()) {
+            Offer offer = modelMapper.map(offerDto, Offer.class);
+            offer.setOwner(currentUser);
+            Offer savedOffer = offerRepository.save(offer);
+            return modelMapper.map(savedOffer, OfferDto.class);
+        } else {
+            throw new NoAccessToOfferOperationException("Access denied, log as Place Owner or System Admin");
```

Comment:
You can simplify this line by calling no-argument constructor of the exception:

```
throw new NoAccessToOfferOperationException()
```

and then by encapsulation your string ```"Access denied, log as Place Owner or System Admin"``` in the exception class:

```
public class NoAccessToOfferOperationException extends RuntimeException {

    public NoAccessToOfferOperationException() {
        super("Access denied, log as Place Owner or System Admin");
    }
}
```

By doing this you can re-use ```NoAccessToOfferOperationException``` exception and throw it in multiple places without remembering to provide the same ```String```.